tests/devices/i11/test_i11devices.py

```diff
@@ -0,0 +1,191 @@
+    set_mock_value(i11_robot.hold, True)  # Set to hold state
+
+    await i11_robot.stage()
+
+    assert await i11_robot.servo_on.get_value() == 1
+    assert await i11_robot.hold.get_value() == 0
+
+
+async def test_robot_pause_resume(i11_robot: NX100Robot) -> None:
+    set_mock_value(i11_robot.hold, False)  # Set to not hold state
+    await i11_robot.pause()
+    assert await i11_robot.hold.get_value() == 1
+
+    set_mock_value(i11_robot.hold, True)  # Set to hold state
+    await i11_robot.resume()
+    assert await i11_robot.hold.get_value() == 0
+
+
+@pytest.mark.parametrize(
+    "state",
+    [
+        RobotSampleState.CAROSEL,
+        RobotSampleState.ONGRIP,
+        RobotSampleState.DIFF,
+    ],
+)
+async def test_when_sample_state_clear_sample_tablein(
+    i11_robot: NX100Robot, state
+) -> None:
+    set_mock_value(i11_robot.robot_sample_state, state)  # Set to not ongrip state
+
+    await i11_robot.clear_sample(table_in=True)
+    assert await i11_robot.job.get_value() == RobotJobs.TABLEIN
+
+
+async def test_robot_clear_sample_when_sample_on_carousel(
+    i11_robot: NX100Robot,
+) -> None:
+    set_mock_value(
+        i11_robot.robot_sample_state, RobotSampleState.CAROSEL
+    )  # Set to not ongrip state
+
+    await i11_robot.clear_sample(table_in=False)
+    assert await i11_robot.robot_sample_state.get_value() == RobotSampleState.CAROSEL
+
+
+async def test_robot_clear_sample_when_sample_unknown(
+    i11_robot: NX100Robot,
+) -> None:
+    set_mock_value(
+        i11_robot.robot_sample_state, RobotSampleState.UNKNOWN
+    )  # Set to not ongrip state
+
+    await i11_robot.clear_sample(table_in=False)
+    assert (
+        not await i11_robot.robot_sample_state.get_value() == RobotSampleState.CAROSEL
+    )
+
+
+async def test_robot_load_sample_when_sample_unknown(i11_robot: NX100Robot) -> None:
+    set_mock_value(
+        i11_robot.robot_sample_state, RobotSampleState.UNKNOWN
+    )  # Set to not ongrip state
+
+    await i11_robot.load_sample(10)  # Load sample at position 10
+    assert not await i11_robot.job.get_value() == RobotJobs.PLACED
+
+
+async def test_robot_clear_load_when_state_is_invalid(i11_robot: NX100Robot) -> None:
+    set_mock_value(i11_robot.robot_sample_state, 99)  # to invalid state
+    with pytest.raises(ValueError):
+        await i11_robot.load_sample(10)  # Load sample at position 10 and fail
+    with pytest.raises(ValueError):
+        await i11_robot.clear_sample()  # clear sample and fail
+
+
+async def test_when_robot_must_stop_and_success_false(
+    i11_robot: NX100Robot,
+) -> None:
+    set_mock_value(
+        i11_robot.robot_sample_state, RobotSampleState.ONGRIP
+    )  # set to ongrip
+    set_mock_value(i11_robot.hold, False)  # Set to not hold state
+    await i11_robot.stop(success=False)
+
+    assert await i11_robot.hold.get_value() == 0
+    assert await i11_robot.job.get_value() == RobotJobs.TABLEIN
+
+
+@pytest.fixture
+async def i11_spinner() -> Spinner:
```

Comment:
Nit: I would normally have one test file per source file but not that big a deal